### DISCUSSION TITLE
Annotations into attributes in PHP 8

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#42584] [Q&A] [answered]
I use Symfony 5.3.6 and PHP 8 .

I want convert annotations into attributes like this:
```
    /**
     * @var string The hashed password
     * @ORM\Column(type="string")
     */
```

But when I convert to this:
```
    #[
        var string The hashed password,
        ORM\Column(type: 'string')
    ]
```

PHPStorm show error!
How can I convert all of this manually?

**Comment by ruudk** [comment, marked as answer]
```
    /**
     * @var string The hashed password
     */
    #[ORM\Column(type: "string")]
```

**Comment by mikepage** [comment]
Rector is the best tool to convert annotations to attributes.

Install `composer require rector/rector --dev`

Add this `rector.php` file in the repository root.

```php
<?php

declare(strict_types=1);

use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $containerConfigurator->import(\Rector\Symfony\Set\SymfonySetList::SYMFONY_52_VALIDATOR_ATTRIBUTES);
    $containerConfigurator->import(\Rector\Doctrine\Set\DoctrineSetList::DOCTRINE_ORM_29);
};
```

Run rector
`./vendor/bin/rector process src/Entity --clear-cache`

**Reply by karaka200** [reply]
Thank you but not work for Controllers.
For example I use
`./vendor/bin/rector process src/Controller/Users --clear-cache`
but not change!

**Reply by mikepage** [reply]
I see, you could add the setlist below for all conversions, including the attributes for Route etc.
 
```php
$containerConfigurator->import(SymfonySetList::SYMFONY_52);
```
